### PR TITLE
[ADF-4614] User preferences in process filters

### PR DESCRIPTION
**Pull request by sivakumar414ram**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://issues.alfresco.com/jira/browse/ADF-4614

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by gmandakini** [comment]
I am also trying to fix the same in one of my pr's but the process-cloud build keeps terminating, even before it comes to the point to run this test :(

File: lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.ts [hunk 1]
```diff
@@ -148,10 +150,20 @@ export class EditProcessFilterCloudComponent implements OnInit, OnChanges {
...
 
     /**
-     * Return process instance filter by application name and filter id
+     * Fetches process instance filter by application name and filter id and creates filter properties, build form
      */
-    retrieveProcessFilter(): ProcessFilterCloudModel {
-        return new ProcessFilterCloudModel(this.processFilterCloudService.getProcessFilterById(this.appName, this.id));
+    retrieveProcessFilterAndBuildForm() {
+        this.isLoading = true;
+        return this.processFilterCloudService.getFilterById(this.appName, this.id)
```
Comment (davidcanonieto) [inline lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.ts]: What are we returning there?

File: lib/process-services-cloud/src/lib/services/user-preference.cloud.service.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      return throwError('Appname not configured');
+    }
+  }
+
+  /**
+   * Creates preference uri
+   * @param appName Name of the target app
+   * @returns String of preference service uri
+   */
+  private buildPreferenceServiceUri(appName: string) {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/services/user-preference.cloud.service.ts]: missing return type

File: lib/process-services-cloud/src/lib/services/user-preference.cloud.service.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+   */
+  updatePreference(appName: string, key: string, updatedPreference: any): Observable<any> {
+    return this.createPreference(appName, key, updatedPreference);
+  }
+
+  /**
+   * Deletes user preference.
+   * @param appName Name of the target app
+   * @param key Key of the target preference
+   * @returns Observable of user preferences
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/services/user-preference.cloud.service.ts]: is it the right description?

File: lib/process-services-cloud/src/lib/services/user-preference.cloud.service.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  /**
+   * Updates user preference.
+   * @param appName Name of the target app
+   * @param key Key of the target preference
+   * @param updatedPreference Details of updated preference
+   * @returns Observable of updated user preferences
+   */
+  updatePreference(appName: string, key: string, updatedPreference: any): Observable<any> {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/services/user-preference.cloud.service.ts]: do we need it if it just proxy the createPreference method?

File: lib/process-services-cloud/src/lib/models/process-services-cloud-query.model.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Pagination } from '@alfresco/js-api';
+
+export interface ProcessServicesCloudResponse {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/models/process-services-cloud-query.model.ts]: Not sure ProcessServicesCloudResponse is the right name, should be more something related to user preference? like UserPreferenceCloudModel

File: lib/process-services-cloud/src/lib/models/process-services-cloud-query.model.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    list?: ProcessServiceCloudEntries;
+}
+
+export interface ProcessServiceCloudEntries {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/models/process-services-cloud-query.model.ts]: UserPreferenceCloudListModel

File: lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.spec.ts [hunk 7]
```diff
@@ -116,6 +117,42 @@ describe('EditProcessFilterCloudComponent', () => {
         });
     });
 
+    it('should not display mat-spinner if isloading set to false', () => {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.spec.ts]: If you use whenStable you need to use the done or async is the same across all the unit test in this file

File: lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.ts [hunk 8]
```diff
@@ -148,10 +150,20 @@ export class EditProcessFilterCloudComponent implements OnInit, OnChanges {
     }
 
     /**
-     * Return process instance filter by application name and filter id
+     * Fetches process instance filter by application name and filter id and creates filter properties, build form
      */
-    retrieveProcessFilter(): ProcessFilterCloudModel {
-        return new ProcessFilterCloudModel(this.processFilterCloudService.getProcessFilterById(this.appName, this.id));
+    retrieveProcessFilterAndBuildForm() {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.ts]: missing return type

**Comment by eromano** [comment]
can you also rebase it ?

File: lib/process-services-cloud/src/lib/models/process-services-cloud-query.model.ts (hunk 5 above)
Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/models/process-services-cloud-query.model.ts]: the response is the same for all cloud services, its not just for the user preferences. for example, all cloud query services return the same structure and it is mapped with "any" type. We could use this model across cloud query services. if "ProcessServicesCloudResponse" name is not sounds good, we can find a better name.

File: lib/process-services-cloud/src/lib/services/user-preference.cloud.service.ts (hunk 2 above)
Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/services/user-preference.cloud.service.ts]: add return type

File: lib/process-services-cloud/src/lib/models/process-services-cloud-query.model.ts (hunk 5 above)
Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/models/process-services-cloud-query.model.ts]: the response is the same for all cloud services, its not just for the user preferences. for example, all cloud query services return the same structure and it is mapped with "any" type. We could use this model across cloud query services. if "ProcessServicesCloudResponse" name does not sound good, we can find a better name.

File: lib/process-services-cloud/src/lib/models/process-services-cloud-query.model.ts (hunk 6 above)
Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/models/process-services-cloud-query.model.ts]: Removed it

**Comment by eromano** [comment]
@sivakumar414ram can you try to rebase the e2e are not running in the process service cloud

**Pull request merged**